src/modules/contacts.js

```diff
@@ -0,0 +1,397 @@
+  constructor() {
+    super()
+    this.date = new Date();
+
+    // Defining variables
+    this.selects = document.getElementsByClassName("contact__add-day");
+    this.years = createNumberArray(1900,this.date.getFullYear());
+    this.database = firebase.database().ref('/contacts');
+    this.months = createNumberArray(1,12);
+    this.days = createNumberArray(1,31);
+    this.birthDays = [];
+    this.unCalled = [];
+    this.inputs = [];
+    this.called = "";
+
+    // Binding functions
+    this.changeContact = this.changeContact.bind(this);
+    this.deleteContact = this.deleteContact.bind(this);
+    this.createContact = this.createContact.bind(this);
+    this.initiateCall = this.initiateCall.bind(this);
+    this.beingAdded = this.beingAdded.bind(this);
+    this.openInfo = this.openInfo.bind(this);
+
+    // Defining intial states
+    this.state = {
+      infoOpened : [0,false],
+      notification : false,
+      unCalled: false,
+      adding : false,
+      filled : true,
+      numbers : []
+    };
+  }
+
+  componentWillMount () {
+    // Fetch data from Firebase
+    this.database.once("value")
+      .then( _ => { 
+        this.setState({numbers : Object.values(_.val())}) 
+
+        // Check if it is someones birthday today
+        Object.values(_.val()).map(obj => obj)
+          .filter(checkBirthDays)
+          .forEach(person => {this.birthDays.push(person.name)})
+        
+        if(this.birthDays.length != 0) this.setState({notification: this.state.notification ? false : true})
+
+        // Check how long it has been since you called someone
+        Object.values(_.val()).map(obj => obj)
+          .filter(checkLastCalled)
+          .forEach(person => this.unCalled.push(person.name))
+
+        if(this.unCalled.length != 0) this.setState({unCalled: this.state.unCalled ? false : true})
+      })
+  } 
+
+  // Update state if contact is being added or added
+  beingAdded () {
+    this.setState({adding: this.state.adding ? false : true});
+  }
+
+  // Creates an object of contact information
+  createContact () {
+    const arrayOfOptions = [...this.selects].map(option => option.options[option.selectedIndex].value);
+    const getInputs = document.querySelectorAll(".contact__add input");
+    const options = checkOptions(arrayOfOptions) ? arrayOfOptions : [];
+    this.inputs = [];
+
+    // Push year to array inputs
+    this.inputs.push(options);
+
+    // Push all onther inputs into array inputs
+    [...getInputs].forEach(input => this.inputs.push(input.value))
+
+    // Check if Name and Number are filled out
+    if(this.inputs[1] === "" || this.inputs[2] === ""){
+      this.setState({filled: false});
+      return
+    }
+
+    // Initialize the addContact function
+    this.addContactFunc("new")
+  }
+
+  // Re-write the numbers object with changed attributes
+  changeContact () {
+    const getInputs = document.querySelectorAll(".contact__info-window__wrapper input");
```

Comment:
Tomu nerozumím. Proč používáš `querySelector`, jaké to má výhody/nevýhody oproti kompozici komponent?

Reply:
Jelikoz jimi pak mohu iterovat s .forEach